### ISSUE TITLE
Node >10.17.0 support

### ISSUE THREAD

**Issue by goszczynskip** [d3/d3-geo-polygon#40]
This package is constrained to node `">=6.0.0 <=10.17.0"`. Using this package with maintained node versions without ignoring *engines* constraint isn't possible. Is there any reason why this constraint is present in *d3-geo-polygon* package.json?

Part of API doesn't seem to rely on node internals eg. *geoArcIntersect*. Maybe if part of API needs old node version it'll be good idea to move independent code parts into other package.

**Comment by Fil** [comment]
(From memory) I think it's because the *tests* involve node-canvas@1 and would have to be rewritten to use node-canvas@2, with it slightly different API.

**Comment by goszczynskip** [comment]
If tests are only concern here than maybe dropping this constraint is an option as short time fix. Adding preinstall step could keep local installs within a 6.x - 10.17.0 range while allowing npm usages.

**Comment by Fil** [comment]
I'd rather move forward to node-canvas@2; the change is minor:

```{diff}
diff --git a/test/render-world b/test/render-world
index fbc4fc4..45a79cd 100755
--- a/test/render-world
+++ b/test/render-world
@@ -10,10 +10,10 @@ if (!/^[a-z0-9]+$/i.test(projectionName)) throw new Error;

 var fs = require("fs"),
     topojson = require("topojson-client"),
-    Canvas = require("canvas"),
+    {createCanvas} = require("canvas"),
     d3 = Object.assign({}, require("d3-geo"), require("../"));

-var canvas = new Canvas(width, height),
+var canvas = createCanvas(width, height),
     context = canvas.getContext("2d");

 var world = require("world-atlas/world/50m.json"),
```

but the idea is to keep in sync with d3-geo and d3-geo-projection, so I'd like to ask @mbostock for his opinion first.

**Comment by Fil** [comment]
A simpler solution at https://github.com/d3/d3-geo-polygon/pull/41#issuecomment-652445847

**Comment by Fil** [comment]
Related: https://github.com/d3/d3-geo-polygon/pull/37

**Comment by Fil** [comment]
fixed by #41
